src/web/rustdoc.rs

```diff
@@ -30,19 +29,23 @@ struct RustdocPage {
     crate_details: Option<CrateDetails>,
 }
 
-impl ToJson for RustdocPage {
-    fn to_json(&self) -> Json {
-        let mut m: BTreeMap<String, Json> = BTreeMap::new();
-        m.insert("rustdoc_head".to_string(), self.head.to_json());
-        m.insert("rustdoc_body".to_string(), self.body.to_json());
-        m.insert("rustdoc_body_class".to_string(), self.body_class.to_json());
-        m.insert("rustdoc_full".to_string(), self.full.to_json());
-        m.insert("rustdoc_status".to_string(), true.to_json());
-        m.insert("name".to_string(), self.name.to_json());
-        m.insert("version".to_string(), self.version.to_json());
-        m.insert("description".to_string(), self.description.to_json());
-        m.insert("crate_details".to_string(), self.crate_details.to_json());
-        m.to_json()
+impl Serialize for RustdocPage {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut state = serializer.serialize_struct("RustdocPage", 9)?;
+        state.serialize_field("rustdoc_head", &self.head)?;
+        state.serialize_field("rustdoc_body", &self.body)?;
+        state.serialize_field("rustdoc_body_class", &self.body_class)?;
+        state.serialize_field("rustdoc_full", &self.full)?;
+        state.serialize_field("rustdoc_status", &true)?;
+        state.serialize_field("name", &self.name)?;
+        state.serialize_field("version", &self.version)?;
+        state.serialize_field("description", &self.description)?;
+        state.serialize_field("crate_details", &self.crate_details)?;
+
+        state.end()
```

Comment:
This looks really similar to what the `derive` would output, the only difference is `&true`, right? I wonder if that would be better as an assert somehow.

Reply:
As above, my goal was a direct translation with testing, leaving the groundwork for myself to come in and make everything sane without breaking it

Reply:
Good point, we can make things simpler in a follow-up PR.